### PR TITLE
feat(sourcemaps): Add sorting to old source maps endpoint

### PR DESCRIPTION
**Pull request by iambriccardo**
This PR adds sorting to the endpoint for listing the release files.

### CODE REVIEW

File: tests/sentry/api/endpoints/test_debug_files.py [hunk 1]
```diff
@@ -323,6 +323,43 @@ def test_source_maps(self):
...
+        )
+
+        release_ids = [release.id, release2.id]
+
+        self.login_as(user=self.user)
+        response = self.client.get(url + "?sortBy=date_added")
+        assert response.status_code == 200, response.content
+        assert list(map(lambda value: value["id"], response.data)) == release_ids
+
+        response = self.client.get(url + "?sortBy=-date_added")
```
Comment (andriisoldatenko) [inline tests/sentry/api/endpoints/test_debug_files.py:359]: add also negative test here with: ` "?sortBy=-added_date"`

File: src/sentry/api/endpoints/debug_files.py [hunk 2]
```diff
@@ -443,10 +443,16 @@ def serialize_results(results):
                 [expose_release(r, file_count_map.get(r["id"], 0)) for r in results], request.user
             )
 
+        sort_by = request.GET.get("sortBy", "-date_added")
+        if sort_by not in {"-date_added", "date_added"}:
```
Comment (andriisoldatenko) [inline src/sentry/api/endpoints/debug_files.py:447]:
```
if sort_by.lstrip("-") != "date_added":
```

Comment (andriisoldatenko) [reply src/sentry/api/endpoints/debug_files.py:447]: YOu can see it works well with both:
```
>>> sort_by = "date_added"
>>> sort_by.lstrip("-")
'date_added'
>>>
```

Comment (iambriccardo) [reply src/sentry/api/endpoints/debug_files.py:447]: I don't like having to hardcore such logic for a such simple thing. I would prefer to be explicit.

Comment (iambriccardo) [reply src/sentry/api/endpoints/debug_files.py:447]: I also feel that way, as we agreed we could follow up with the serializers and all the proper implementations.

**Review by andriisoldatenko — APPROVED** [review]

**Pull request merged**